pkg/credentials/filecache.go

```diff
@@ -173,6 +191,15 @@ func writeCache(filename string, cache cacheFile) error {
 // otherwise fetching the credential from the underlying Provider and caching the results on disk
 // with an expiration time.
 func (f *FileCacheProvider) Retrieve() (credentials.Value, error) {
+	f.once.Do(func() {
```

Comment:
Why is this needed?

Reply:
When the credentials aren't cached, both caches will try to obtain the MFA token from STDIN, prompting the user twice. This check is for obtaining the credentials from the file cache, only for the first call to Retrieve to ensure it reads the cached token if it was set by the other cache.